### PR TITLE
[Relay][Frontend][ONNX] Fix reshape precompute, and type error

### PR DESCRIPTION
**Pull request by jroesch**
Fix the code which precomputes shapes, and repair type error by casting literal integers to int32.

### CODE REVIEW

**Comment by jroesch** [comment]
cc @tqchen @zhreshold

**Comment by tqchen** [comment]
Please fix the lint error

**Comment by tqchen** [comment]
@jroesch can you look into the CI error?

**Comment by jroesch** [comment]
I disabled a few tests that require debugging version issues on CI. I propose we merge this, and I'll do a follow up pass afterwards. 

See my recent issue #3247.

**Comment by zhreshold** [comment]
The CI dependency version issue is really a headache now.

**Comment by jroesch** [comment]
Yeah this is annoying, @zhreshold  @tqchen can we bump the ONNX version?

**Comment by tqchen** [comment]
Ideally, we should make the code compatible with both versions. Please send a PR to https://github.com/dmlc/tvm/blob/master/docker/install/ubuntu_install_onnx.sh#L24 to ping the onnx to a specific version. then we can try to upgrade the image

**Comment by jroesch** [comment]
> Ideally, we should make the code compatible with both versions. Please try to do that and let us merge it in. Please send a PR to https://github.com/dmlc/tvm/blob/master/docker/install/ubuntu_install_onnx.sh#L24 to ping the onnx to a specific version. then we can try to upgrade the image

The problem is that the operator support in ONNX and in the torch export functionality varies across versions, so tests have to be conditionally enabled if we want to test model support. I believe it is important we test the entire pipeline on our CI because these models seem to have a lot of failures on the discuss board lately.

**Comment by aa12356jm** [comment]
i also  encounter this problem,looking forward to solving it quickly,thanks

**Comment by tqchen** [comment]
https://github.com/dmlc/tvm/pull/3374 updates the docker image to the latest state

File: tests/python/frontend/onnx/test_forward.py [hunk 1]
```diff
@@ -1070,6 +1073,47 @@ def test_LogSoftmax():
...
+    torch.onnx.export(model(), dummy_input, file_name, export_params=True, verbose=False)
+    onnx_model = onnx.load(file_name)
+    shapes = { '0' : input_size }
+    expr, params = relay.frontend.from_onnx(onnx_model, shape=shapes)
+
+def test_resnet():
+    check_torch_conversion(torchvision.models.resnet18, (1,3,224,224))
+    # check_torch_conversion(torchvision.models.resnet101, (1,3,224,224))
+
+# def test_alexnet():
```
Comment (tmoreau89) [inline tests/python/frontend/onnx/test_forward.py]: What's the reason for these tests being commented out? Can we add a TODO if it's pending on some feature that isn't implemented yet?

**Comment by tqchen** [comment]
Thanks @jroesch @zhreshold this PR has been merged. I need to fix issues in #3374 and will report back when that is ready

**Comment by tqchen** [comment]
Note: I reverted this PR for now due to CI errors, please open a new PR to bring things back, sorry about the trouble.

**Comment by zhiics** [comment]
The problem is because fusion was not performed here:
https://github.com/dmlc/tvm/pull/3230/files#diff-03149f7671cff8be6734838f7707a24dR417

**Comment by tqchen** [comment]
NOTE, now that https://github.com/dmlc/tvm/pull/3374 has been merged, you should be able to use the latest env

**Pull request merged**